packages/kestrel_core/src/kestrel/session.py

```diff
@@ -90,16 +95,19 @@ def evaluate_instruction(self, ins: Instruction) -> Display:
         while True:
             for g in self.irgraph.find_dependent_subgraphs_of_node(ins, _cache):
                 interface = _interface_manager[g.interface]
+                _logger.debug("eval: subgraph: %s", [i.instruction for i in g.nodes()])
+                _logger.debug("eval: interface = %s", interface)
                 for iid, _display in (
                     interface.explain_graph(g)
                     if is_explain
-                    else interface.evaluate_graph(g)
+                    else interface.evaluate_graph(g, _cache)
                 ).items():
                     if is_explain:
                         display.graphlets.append(_display)
+                        if interface is not _cache:
```

Comment:
Shall we also remove the if statement like the evaluation case?

Reply:
I had to move this line here otherwise the explain unit tests failed - the `Construct` node would be omitted from the explain output, though I'm not completely certain why.

Reply:
It is correct to omit the `Construct` node since it is already evaluated in the 3rd subgraph (`DISP domain`). The results of `domain` should be cached, preventing the 4th subgraph to trace back to the `Construct`.

Did a patch commit for it.